aws/rust-runtime/aws-config/src/lib.rs

```diff
@@ -468,6 +470,60 @@ mod loader {
             self
         }
 
+        #[doc(hidden)]
+        /// Sets an [`Interceptor`] that runs at specific stages of the request execution pipeline.
+        /// This takes effect for **all** AWS services.
+        ///
+        /// Interceptors targeted at a certain stage are executed according to the pre-defined priority.
+        /// SDK provides the default set of interceptors. An interceptor configured by this method
+        /// will run after those default interceptors.
+        ///
+        /// ## Examples
+        /// ```no_run
+        /// # #[cfg(feature = "examples")]
+        /// # fn example() {
+        /// use aws_http::user_agent::AwsUserAgent;
+        /// use aws_smithy_runtime_api::client::interceptors::{Interceptor, InterceptorContext};
+        /// use aws_smithy_runtime_api::config_bag::ConfigBag;
+        /// use aws_types::sdk_config::{Builder, SdkConfig};
+        /// use http::header::USER_AGENT;
+        /// use http::{HeaderName, HeaderValue};
+        ///
+        /// #[derive(Debug)]
+        /// struct TestUserAgentInterceptor;
```

Comment:
I'm a little hesitant to have this one as an example since changing the user agent should be discouraged. Is there another one that illustrates it?

Reply:
Thanks for calling it out. To be honest, I was not entirely sure about the usefulness of this example, plus it causes issues as we know:
- Including `AwsUserAgent` in an example for generic clients (as pointed out [here](https://github.com/awslabs/smithy-rs/pull/2669#discussion_r1184309128))
- `aws-types` having to depend on `aws-http` to be aware of the type `AwsUserAgent` so the example can compile ([relevant failure](https://github.com/awslabs/smithy-rs/actions/runs/4875403403/jobs/8697568880) in CI)
- Showing an example of user workflow that's discouraged

Coming up with a more appropriate example should solve all these issues. Will update it in a subsequent commit.

Reply:
I've replaced the example with the one modifying a request URI in 5583680. This is based on a customer's actual use case. Let me know if we should present a different example.